### ISSUE TITLE
Default arguments cannot be set from constants

### ISSUE THREAD

**Issue by michwill** [vyperlang/vyper#1627]
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b12+commit.a01cdc8
* OS: linux
* Python Version (output of `python --version`): 3.6.8
* Environment (output of `pip freeze`):

### What's your issue about?

Compilation of this fails. Curiously enough, it works if `a: uint256 = [Z, Z]`.
```python
from vyper import compile_code
compile_code("""
Z: constant(uint256) = 0
A: constant(uint256[2]) = [Z, Z]

@public
def g(a: uint256[2] = A):
    pass
""")
```

**Comment by iamdefinitelyahuman** [comment]
https://github.com/ethereum/vyper/pull/1607 implemented support for builtin constants as default args, but not user-defined constants.

`a: uint256 = [Z, Z]` compiling is related to #1628, I'd guess.

**Comment by fubuloubu** [comment]
Confirm the following compiles:
```python
Z: constant(uint256) = 0
A: constant(uint256[2]) = [Z, Z]
```

Confirm the following does not compile:
```python
Z: constant(uint256) = 0
A: constant(uint256[2]) = [Z, Z]

@public
def g(a: uint256[2] = A):
    pass
```

**Comment by iamdefinitelyahuman** [comment]
Fixed by type checking in #2029
